src/Compilers/Core/Portable/Symbols/Nullability.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Enumeration of the possible nullability states for types.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Generally, <see cref="Nullability"/> is used both for declared nullability
+    /// and for inferred nullability. The declared nullability of a symbol is what
+    /// the code declared when the variable was defined. For example:
+    /// </para>
+    /// <para>
+    /// <code>
+    /// string s;
+    /// </code>
+    /// This declares a variable of type <code>String</code> with nullability <see cref="NonNull"/>,
+    /// assuming that this is in C# 8 code with a <code>NonNullTypes</code> attribute.
+    /// </para>
+    /// <para>
+    /// <code>
+    /// string? s;
+    /// </code>
+    /// This declares a variable of type <code>String</code> with nullability <see cref="MaybeNull"/>.
+    /// </para>
+    /// <para>
+    /// The compiler also tracks inferred nullability. For example:
+    /// <code>
+    /// object? o1 = "Hello world!";
+    /// object? o2 = null;
+    /// o1.ToString();
+    /// o2.ToString(); // Warning reported
+    /// if (o2 != null) o2.ToString(); // No warning reported
+    /// </code>
+    /// <code>o1</code> is inferred to be <see cref="NonNull"/>, despite being declared
```

Comment:
><code>o1</code> is inferred to be <see cref="NonNull"/> [](start = 8, length = 55)

This should be clarified; it is not the *variable* that is inferred to be `NonNull`, but the use of the variable in its context.